In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("data\diary.csv")

In [4]:
df.describe()

,Year,Rating
count,320.000000,285.000000
mean,2013.209375,3.926316
std,12.016312,0.613672
min,1957.000000,0.500000
25%,2004.000000,3.500000
50%,2018.000000,4.000000
75%,2023.000000,4.500000
max,2026.000000,5.000000


## YEAR ##
*Since i started using letterboxd i have put 320 movies in my diary
*The oldest movie i have watched was in 1957 while the newest one was in 2026 
## RATINGS ##
*I have rated 285 movies 
*The minimum rating i have given a movie is 0.5/5 stars
*The maximum i have given a movie is 5/5 stars
*my average movie rating out of the 285 rated is 3.9/5 stars 

In [12]:
df.info

<bound method DataFrame.info of            Date                                               Name  Year  \
0    2024-08-19                                                 Us  2019   
1    2024-08-20                                      6 Underground  2019   
2    2024-08-20  Birds of Prey (and the Fantabulous Emancipatio...  2020   
3    2024-08-19                                               Dune  2021   
4    2024-08-20                                           Free Guy  2021   
..          ...                                                ...   ...   
315  2026-06-14                                     Disclosure Day  2026   
316  2026-06-24                                             Poison  2023   
317  2026-06-25                                    I Love Boosters  2026   
318  2026-06-26                       Batman: Mask of the Phantasm  1993   
319  2026-06-29                                          Happiness  1998   

             Letterboxd URI  Rating Rewatch            

In [10]:
df.columns

Index(['Date', 'Name', 'Year', 'Letterboxd URI', 'Rating', 'Rewatch', 'Tags',
       'Watched Date'],
      dtype='str')

In [ ]:
max_repetitions = df['Watched Date'].value_counts().max()
most_frequent_item = df['Watched Date'].value_counts().idxmax()
print(f"On the day '{most_frequent_item}' i watched {max_repetitions} films the most ever since i started using letterboxd ")

On the day '2025-08-02' i watched 7 films the most ever since i started using


In [29]:
rewatched_movie_names = df[df['Rewatch'].isin(['Yes', True, 'true'])]['Name'].unique()
rewatch_counts = df[df['Name'].isin(rewatched_movie_names)]['Name'].value_counts()
rewatch_summary = rewatch_counts.reset_index()
rewatch_summary.columns = ['Movie Title', 'Total Times Rewatched']
rewatch_summary.head(20)

,Movie Title,Total Times Rewatched
0,Dune,2
1,Mad Max: Fury Road,2
2,Furiosa: A Mad Max Saga,2
3,Arrival,2
4,Talk to Me,2
5,Game Night,2
6,Sinners,2
7,Whiplash,1
8,(500) Days of Summer,1
9,Bullet Train,1


## REWATCH ##
These are the number of times i have rewatched those movies since i started using letterboxd im a real rewatch-aholic

